src/wasm-features.h

```diff
@@ -134,7 +139,25 @@ struct FeatureSet {
   void setTypedFunctionReferences(bool v = true) {
     set(TypedFunctionReferences, v);
   }
-  void setAll(bool v = true) { features = v ? All : MVP; }
+  void setGCNNLocals(bool v = true) { set(GCNNLocals, v); }
+  void setAll(bool v = true) {
+    // Do not set GCNNLocals, which forces the user to opt in to that feature
+    // explicitly. That is, wasm-opt -all will enable GC but *not* enable
+    // non-nullable locals. To get them, do wasm-opt -all --enable-gc-nn-locals
+    // FIXME: When the wasm spec stabilizes, this feature will go away, as the
+    //        non-nullable locals experiment will either become the standard,
+    //        or it will go away.
+    if (v) {
+      // Leave the old GCNNLocals value unmodified. This makes things like
+      // --enable-gc-nn-locals -all work (that is, if we enable the feature,
+      // then -all does not disable it; it simply does not enable it by itself).
+      auto oldGCNNLocals = hasGCNNLocals();
+      features = All;
+      setGCNNLocals(oldGCNNLocals);
+    } else {
+      features = MVP;
```

Comment:
What does `setAll(false)` mean? I couldn't find that usage anywhere. If that is the same as `--mvp-features`, `--enable-gc-nn-locals --mvp-features` does not enable gc-nn-locals and it is intended, right?

Reply:
Yes, I'm not sure either why `setAll()` takes a parameter - for symmetry maybe? But it does seem like `setAll(true)` is the only use case...

Yes, `--enable-X --mvp-features` would disable X, effectively. The new feature behaves the same as all the others (unless I have a bug here).

Reply:
I'm surprised that the `false` case still unsets the new feature given that the `true` case does not set it. But if we have no users of the `false` case, perhaps the better solution would be to just remove the parameter.